### PR TITLE
basic methods implementations and testing bug fix

### PR DESCRIPTION
**Pull request by Krystal-158**
1. Adds supplierID(int) to data schema in model.py and change the type of productID to integer
2. To keep code simple, productIDList is changed into an integer list instead of a list of table references.
3. Implements add(), serialize() and deserialize() methods in model.py
4. Add init to make the whole folder a python package named 'service

### CODE REVIEW

**Comment by Krystal-158** [comment]
When I test the serialize() in model.py an exception is always caught, I still have no idea about this. This is how I do this:
1. uncomments line 107 supplier.deserialize(data) in service.py.
2. git pull->restart virtual box-> cd /vagrant ->FLASK_APP=service:app flask run -h 0.0.0.0
3. then in postman type in "POST http://0.0.0.0:2333/suppliers" and in body input 'supplier:ID'
￼
屏幕快照 2019-10-13 19 16 43

4. will see error 400: Invalid supplier: body of request contained bad or no data

File: service/models.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+class Supplier(db.Document):
+    """
+    Suppliers data schema: https://github.com/nyu-devops-fall19-suppliers/suppliers/issues/21
+    """
+    logger = logging.getLogger('flask.app')
+    app = None
+
+    # Table Schema
+    supplierID = db.IntField(required=True)
```
Comment (greenson) [inline service/models.py]: I think MongoDB would automatically generate a unique **_id** field for each collection, like **4f4381f4e779897a2c000009**. Thus I believe we don't need manually assign an integer to each supplier.   

Here is a stackoverflow page on [How to search document by oid in mongoengine](https://stackoverflow.com/questions/9988352/how-to-search-document-by-oid-in-mongoengine)

File: service/models.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
+import logging
+from flask import Flask
+from flask_mongoengine import MongoEngine
+from flask_mongoengine.wtf import model_form
+
+db = MongoEngine()
```
Comment (greenson) [inline service/models.py]: I think **MongoEngine()** should accept a parameter **app** indicating which DB to connect?

**Comment by silemie** [comment]
One question:
It seems that the create operation has some error. 
When I request POST http://127.0.0.1:2333/suppliers, it will show 505 Internal Server Error. 
It shows that NameError: name 'check_content_type' is not defined

**Pull request merged**